src/services/array-functions.js

```diff
@@ -14,27 +25,41 @@ export function map(theArray, fnc){
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
+
 export function filter(theArray, fnc){
-    return theArray.filter(x => x.length === 3)
-}
+    let newArray = [];
+    for(let i =0; i < theArray.length; i ++){
+        if(fnc(theArray[i])){
+            newArray.push(theArray[i]);
+        }
+    }
+        
+    return newArray;
+    }
+
 
 
 //loop theArray and call the fnc for each thing in the array, 
 //     passing in the item from the current loop
 //fnc will return true or false, if true return the item 
 //return null
 export function find(theArray, fnc){
+    for(let i = 0; i < theArray.length; i ++){
```

Comment:
I think I've sorted it out:

export function find(theArray, fnc){
    for(let i = 0; i < theArray.length; i ++){
        if(fnc(theArray[i])){
            return theArray[i];
        }
}
seems to work